### ISSUE TITLE
Import conflict in older ghc versions.

### ISSUE THREAD

**Issue by mdorman** [seagreen/hjsonschema#22]
If you check out the [Travis-CI runs](https://travis-ci.org/mdorman/couch-simple/builds/84801828) runs for my changes to [couch-simple](http://github.com/mdorman/couch-simple) to support 0.7.0.0 (in mdorman/couch-simple@4b16fe15962ca274d4473eaca6fb7c12f102ef34), you can see older versions of ghc are failing to build hjsonschema, with the error:

```
[ 3 of 12] Compiling Data.JsonSchema.Core ( src/Data/JsonSchema/Core.hs, dist/build/Data/JsonSchema/Core.o )
src/Data/JsonSchema/Core.hs:24:21:
    Ambiguous occurrence `concat'
    It could refer to either `Prelude.concat',
                             imported from `Prelude' at src/Data/JsonSchema/Core.hs:3:8-27
                             (and originally defined in `GHC.List')
                          or `Import.concat',
                             imported from `Import' at src/Data/JsonSchema/Core.hs:9:1-23
                             (and originally defined in `Data.Foldable')
```

I'm guessing that this is because in ghc-7.10, they're both using the same implementation of `concat`, but in older versions of ghc they were distinct.

**Comment by seagreen** [comment]
Is this a priority for you @mdorman? I'm having trouble supporting 7.8 without making real changes to the code.

Right now I've got:

```
fetchReferencedSchemas'
  :: forall t e m err. (IsString t, MonadError t e, Traversable e, MonadIO m)
  => (Text -> m (e LBS.ByteString)) -- (Provide your own fetching function)
  -> Spec err
  -> RawSchema
  -> Graph
  -> m (e Graph) -- (Choose your own error type)
```

Which is reified for normal use cases as:

```
fetchReferencedSchemas :: Spec err -> RawSchema -> Graph -> IO (Either Text Graph)
```

But I can't do the reification without a Traversable instance for Either.

**Comment by dylex** [comment]
I'm stuck with 0.6 as well until this is fixed, but I can get 0.7 it to build fine on 7.8 with some simple changes. base 4.7 does provide a Traversable Either instance, but you do need an extra Functor m constraint on that definition.

**Comment by seagreen** [comment]
Thanks for the advice guys! New version released that compiles on GHC 7.8; pushed to Hackage.

**Comment by seagreen** [comment]
PS @mdorman you'll be happy to know that I've got Travis CI setup again:)
